src/main/java/com/ldtteam/structures/lib/BlueprintTagUtils.java

```diff
@@ -43,4 +45,59 @@ public static BlockPos getFirstPosForTag(final Blueprint blueprint, final String
 
         return null;
     }
+
+    /**
+     * Gets the number of layers at the bottom of the blueprint that are considered
+     * 'ground' (or water, for ships).
+     * @param blueprint The blueprint to query
+     * @param defaultGroundLevels The number of levels to assume if there is no tag
+     * @return The number of levels, e.g. 1 means that the bottom layer of the
+     *         blueprint is ground and 3 means the bottom 3 layers are ground.
+     */
+    public static int getNumberOfGroundLevels(@NotNull final Blueprint blueprint,
+                                              final int defaultGroundLevels)
+    {
+        final BlockPos groundLevelPos = getFirstPosForTag(blueprint, GROUNDLEVEL_TAG);
+        if (groundLevelPos != null)
+        {
+            return blueprint.getPrimaryBlockOffset().getY() + groundLevelPos.getY() + 1;

```

Comment:
This one seems weird to me: E.g. we got the anchor/prim offset at Y=5, the tag Y-level is at 7 and then the groundlevels are from 0 to 13?

Reply:
The tag y-level is relative to the anchor.  So `groundLevelPos.getY() == -1` where the ground is one level below the anchor, for example.